### PR TITLE
 BUG/ENH: Handle AmbiguousTimeError in date rounding

### LINKED ISSUE #18946: AmbiguousTimeError in floor() operation
This is probably related to #18885 

Given a DataFrame with a DST change on it:

```python
df1=pd.DataFrame([pd.to_datetime('2017-10-29 02:00:00+02:00'),pd.to_datetime('2017-10-29 02:00:00+01:00'),pd.to_datetime('2017-10-29 03:00:00+01:00')],columns=['date'])
df1['date'] = df1['date'].dt.tz_localize('UTC').dt.tz_convert('Europe/Madrid')
df1['value'] = 1
```

When we try to do a `floor` operation, we get an AmbiguousTimeError exception:

```python
df1.date.dt.floor('H')
```

### Expected output

```
0   2017-10-29 02:00:00+02:00
1   2017-10-29 02:00:00+01:00
2   2017-10-29 03:00:00+01:00
```

### Actual output

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "(...)/venv/lib/python3.6/site-packages/pandas/core/accessor.py", line 115, in f
    return self._delegate_method(name, *args, **kwargs)
  File "(...)/venv/lib/python3.6/site-packages/pandas/core/indexes/accessors.py", line 131, in _delegate_method
    result = method(*args, **kwargs)
  File "(...)/venv/lib/python3.6/site-packages/pandas/core/indexes/datetimelike.py", line 118, in floor
    return self._round(freq, np.floor)
  File "(...)/venv/lib/python3.6/site-packages/pandas/core/indexes/datetimelike.py", line 110, in _round
    self._shallow_copy(result, **attribs))
  File "(...)/venv/lib/python3.6/site-packages/pandas/core/indexes/datetimelike.py", line 230, in _ensure_localized
    result = result.tz_localize(self.tz)
  File "(...)/venv/lib/python3.6/site-packages/pandas/util/_decorators.py", line 118, in wrapper
    return func(*args, **kwargs)
  File "(...)/venv/lib/python3.6/site-packages/pandas/core/indexes/datetimes.py", line 1858, in tz_localize
    errors=errors)
  File "pandas/_libs/tslib.pyx", line 3593, in pandas._libs.tslib.tz_localize_to_utc
pytz.exceptions.AmbiguousTimeError: Cannot infer dst time from Timestamp('2017-10-29 02:00:00'), try using the 'ambiguous' argument
```

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #18946
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Discussed in #22560, adding an `ambiguous` argument to `round`, `ceil` and `floor` so the user can dictate how to handle rounding timestamps that land on ambiguous times.

### CODE REVIEW

File: pandas/_libs/tslibs/nattype.pyx [hunk 1]
```diff
@@ -490,6 +495,15 @@ class NaTType(_NaT):
         Parameters
         ----------
         freq : a freq string indicating the flooring resolution
+        ambiguous : bool, 'NaT', default 'raise'
+            - bool contains flags to determine if time is dst or not (note
+              that this flag is only applicable for ambiguous fall dst dates)
+            - 'NaT' will return NaT for an ambiguous time
```
Comment (jreback) [inline pandas/_libs/tslibs/nattype.pyx]: can you add versionadded tags here (and other new arg places)

File: pandas/_libs/tslibs/nattype.pyx [hunk 2]
```diff
@@ -490,6 +495,15 @@ class NaTType(_NaT):
...
+            - 'raise' will raise an AmbiguousTimeError for an ambiguous time
+
```
Comment (jreback) [inline pandas/_libs/tslibs/nattype.pyx]: may make sense to provide a template for the doc-strings (can be followup)

File: pandas/core/indexes/datetimelike.py [hunk 3]
```diff
@@ -99,6 +99,16 @@ class TimelikeOps(object):
             frequency like 'S' (second) not 'ME' (month end). See
             :ref:`frequency aliases <timeseries.offset_aliases>` for
             a list of possible `freq` values.
+        ambiguous : 'infer', bool-ndarray, 'NaT', default 'raise'
+            - 'infer' will attempt to infer fall dst-transition hours based on
+              order
+            - bool-ndarray where True signifies a DST time, False designates
+              a non-DST time (note that this flag is only applicable for
```
Comment (jreback) [inline pandas/core/indexes/datetimelike.py]: versionadded

can you test for .dt accessors as well?

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine

File: pandas/core/indexes/datetimelike.py (hunk 3 above)
Comment (mroeschke) [reply pandas/core/indexes/datetimelike.py]: The .dt accessor tests are here: https://github.com/pandas-dev/pandas/blob/ef263112fa9e7f8ac8a3b965fab7fb111adace20/pandas/tests/series/test_datetime_values.py#L234

**Comment by mroeschke** [comment]
Addressed your comments and all green @jreback

**Review by jreback — APPROVED** [review]

**Pull request merged**